src/connector/sqlite/conversion.rs

```diff
@@ -1,73 +1,131 @@
-                    Some(n) if n.starts_with("CHARACTER") => Value::Text(None),
-                    Some(n) if n.starts_with("character") => Value::Text(None),
-                    Some(n) if n.starts_with("VARCHAR") => Value::Text(None),
-                    Some(n) if n.starts_with("varchar") => Value::Text(None),
-                    Some(n) if n.starts_with("VARYING CHARACTER") => Value::Text(None),
-                    Some(n) if n.starts_with("varying character") => Value::Text(None),
-                    Some(n) if n.starts_with("NCHAR") => Value::Text(None),
-                    Some(n) if n.starts_with("nchar") => Value::Text(None),
-                    Some(n) if n.starts_with("NATIVE CHARACTER") => Value::Text(None),
-                    Some(n) if n.starts_with("native character") => Value::Text(None),
-                    Some(n) if n.starts_with("NVARCHAR") => Value::Text(None),
-                    Some(n) if n.starts_with("nvarchar") => Value::Text(None),
-                    Some(n) if n.starts_with("DECIMAL") => Value::Real(None),
-                    Some(n) if n.starts_with("decimal") => Value::Real(None),
-                    Some("BLOB") => Value::Bytes(None),
-                    Some("blob") => Value::Bytes(None),
-                    Some("NUMERIC") | Some("REAL") | Some("DOUBLE") | Some("DOUBLE PRECISION") | Some("FLOAT") => {
-                        Value::Real(None)
-                    }
-                    Some("numeric") | Some("real") | Some("double") | Some("double precision") | Some("float") => {
-                        Value::Real(None)
-                    }
-                    Some("DATE") | Some("DATETIME") => Value::DateTime(None),
-                    Some("date") | Some("datetime") => Value::DateTime(None),
-                    Some("BOOLEAN") => Value::Boolean(None),
-                    Some("boolean") => Value::Boolean(None),
-                    Some(n) => panic!("Value {} not supported", n),
-                    None => Value::Integer(None),
+                ValueRef::Null => match column {
+                    c if c.is_integer() | c.is_null() => Value::Integer(None),
+                    c if c.is_text() => Value::Text(None),
+                    c if c.is_bytes() => Value::Bytes(None),
+                    c if c.is_real() => Value::Real(None),
+                    c if c.is_datetime() => Value::DateTime(None),
+                    c if c.is_date() => Value::Date(None),
+                    c if c.is_bool() => Value::Boolean(None),
+                    c => match c.decl_type() {
+                        Some(n) => {
+                            let msg = format!("Value {} not supported", n);
+                            let kind = ErrorKind::conversion(msg);
+
+                            Err(Error::builder(kind).build())?
+                        }
+                        None => Value::Integer(None),
+                    },
```

Comment:
much better!

Reply:
Yeah. Not perfect, but at least better...